node_modules/oae-preview-processor/tests/test-previews.js

```diff
@@ -910,15 +910,82 @@ describe('Preview processor', function() {
+            nock.enableNetConnect();
+
+            // Expect GET requests to:
+            // https://www.googleapis.com/youtube/v3/videos?part=snippet&id=...&key=...
+            nock('https://www.googleapis.com')
+                .get('/youtube/v3/videos?part=snippet&id=lgTQ5I_H4Xk&key=')
+                .twice()
+                .reply(200, {
+                    'kind': 'youtube#videoListResponse',
+                    'etag': '\"tbWC5XrSXxe1WOAx6MK9z4hHSU8/CQhw2t_NZKBaw72WEH7b1hSa6RA\"',
+                    'pageInfo': {
+                        'totalResults': 1,
+                        'resultsPerPage': 1
+                    },
+                    'items': [
+                        {
+                            'kind': 'youtube#video',
+                            'etag': '"tbWC5XrSXxe1WOAx6MK9z4hHSU8/TScJmbzIomSQvSDFqAgTXGr7Y2U"',
+                            'id': 'lgTQ5I_H4Xk',
+                            'snippet': {
+                                'publishedAt': '2013-05-09T20:54:56.000Z',
+                                'channelId': 'UCzDbnWaP_5kd6HpvDUjoT4Q',
+                                'title': 'How to prounounce \"Apereo\"',
+                                'description': 'Here is Ian Dolphin, the Executive Director of the Apereo Foundation, with the official pronunciation of the word \"Apereo\".',
+                                'thumbnails': {
+                                    'default': {
+                                        'url': 'https://i.ytimg.com/vi/lgTQ5I_H4Xk/default.jpg',
+                                        'width': 120,
+                                        'height': 90
+                                    },
+                                    'medium': {
+                                        'url': 'https://i.ytimg.com/vi/lgTQ5I_H4Xk/mqdefault.jpg',
+                                        'width': 320,
+                                        'height': 180
+                                    },
+                                    'high': {
+                                        'url': 'https://i.ytimg.com/vi/lgTQ5I_H4Xk/hqdefault.jpg',
+                                        'width': 480,
+                                        'height': 360
+                                    }
+                                },
+                                'channelTitle': 'Apereo Foundation',
+                                'categoryId': '28',
+                                'liveBroadcastContent': 'none',
+                                'localized': {
+                                    'title': 'How to prounounce \"Apereo\"',
+                                    'description': 'Here is Ian Dolphin, the Executive Director of the Apereo Foundation, with the official pronunciation of the word \"Apereo\".'
+                                }
+                            }
+                        }
+                    ]
+                });
+        };
```

Comment:
That's what we used to do originally. But APIs like slideshare or Flickr were intermittently failing the builds due to network issues or just being down in general (https://github.com/oaeproject/Hilary/pull/915)